Day 13: “A11y Considerations in Math on the Web” / Manuel Sánchez (#204)
matuzo/HTMHell: hell/adventcalendar/2025/13/index.md

---
title: "A11y Considerations in Math on the Web"
author: "Manuel Sánchez"
author_bio: "Accessibility specialist and game developer at [DIE ZEIT](https://www.zeit.de/spiele/index). Currently creating [The Runic Edda](https://www.therunicedda.com), a solo game blending storytelling, Viking history, and fun. I am driven by curiosity, creativity, and the belief that technology should welcome everyone."
date: 2025-12-13
author_links:
  - label: "Web"
    url: "https://www.manuelsanchezdev.com"
    link_label: "manuelsanchezdev.com"
  - label: "Bluesky"
    url: "https://bsky.app/profile/manuelsanchezdev.com"
    link_label: "@manuelsanchezdev.com"
intro: "<p>Math on the web has always been a visual and accessibility challenge. In this article, we learn how to make structures understandable for assistive technologies by using today's MathML Core.</p>"
image: "advent25_13"
---

Maybe it has happened to you that you wanted to write some formulas in HTML to display them on a website, and even though there are multiple ways to do it, accessibility is often not considered in the process. How the formula is read by screen readers is crucial to ensure that we don't leave anyone behind.

> Even though the web is full of many different and interesting approaches, the reality is that [MathML](https://developer.mozilla.org/en-US/docs/Web/MathML) is usually the best option for this problem. It has its own syntax and was not originally designed for the web, but MathML provides various elements that give the correct semantics to the different parts of a formula.

Reviewer: > MathML is usually the best option

Your article is about accessibility, so I have to assume that this statement also implies that it's the best option in terms of accessibility. Is this true? You've included a screenshot of the accessibility tree, but what about actual screen reader support? Have you tested this with common SR/browsers combinations?

Author: Hi! Yes, it was a bit vague formulated on my side! I wrote it in another way and gave some examples of other approaches following the suggestion of @ZoeBijl. I just mentioned here it has good screen reader support but will deepen later with some examples once we test the Pythagorean theorem.

> Even though the web is full of many different and interesting approaches, the reality is that [MathML](https://developer.mozilla.org/en-US/docs/Web/MathML) is usually the best option for this problem. It has its own syntax and was not originally designed for the web, but MathML provides various elements that give the correct semantics to the different parts of a formula.

Author: Sure, good idea!

> Even though the web is full of many different and interesting approaches, the reality is that [MathML](https://developer.mozilla.org/en-US/docs/Web/MathML) is usually the best option for this problem. It has its own syntax and was not originally designed for the web, but MathML provides various elements that give the correct semantics to the different parts of a formula.

Author: Good idea! Just wrote some more lines about that!

Let's take the famous Pythagorean Theorem as an example.

> <section style="margin-bottom: 2rem" >

Reviewer 2: would it make sense to follow your own advise and use an `aria-label` here?

  <math xmlns="http://www.w3.org/1998/Math/MathML">
    <msup>
      <mi>a</mi>
      <mn>2</mn>
    </msup>
    <mo>+</mo>
    <msup>
      <mi>b</mi>
      <mn>2</mn>
    </msup>
    <mo>=</mo>
    <msup>
      <mi>c</mi>
      <mn>2</mn>
    </msup>
  </math>
</section>

```html
<math xmlns="http://www.w3.org/1998/Math/MathML">
  <msup>
    <mi>a</mi>
    <mn>2</mn>
  </msup>
  <mo>+</mo>
  <msup>
    <mi>b</mi>
    <mn>2</mn>
  </msup>
  <mo>=</mo>
  <msup>
    <mi>c</mi>
    <mn>2</mn>
  </msup>
</math>
```

> Unlike plain HTML with <code>sup</code> or <code>span</code>, which only describe presentation, MathML defines each role explicitly:

Reviewer 2: this is a good selling point for MathML. are there other solutions that offer this functionality too? or is this exclusive to MathML?

Author: I think yes, other options have this functionality too by using MathML in a hidden layer somehow (Assistive/hidden MathML layer)
The biggest selling point is then that MathML it’s the only web-standard markup that natively expresses math roles in the DOM.

Author: Do you think I should mention that?

Reviewer 2: > Do you think I should mention that?

I think that is a good idea :)

- <code>math</code> represents the entire mathematical expression.
- <code>msup</code> defines a superscript relationship (a base and an exponent).
- <code>mi</code> is a mathematical identifier, typically a variable such as <strong>a</strong>, <strong>b</strong>, or <strong>c</strong>.
- <code>mn</code> is a mathematical number, like <strong>2</strong>.
- <code>mo</code> is a mathematical operator, such as <strong>+</strong> or <strong>=</strong>.

> <p class="highlight"><strong>Note:</strong> This is only one way to display the formula. If you're interested in how to actually prove it, Mozilla has a very detailed page about <a href="https://developer.mozilla.org/en-US/docs/Web/MathML/Guides/Proving_the_Pythagorean_theorem">proving the Pythagorean theorem with MathML</a>.</p>

Author: Done!

Anyway, with this approach, the accessibility tree shows good semantics.

<img alt="Accessibility tree view of a MathML formula showing nested semantic elements. The tree includes nodes such as MathMLMath, MathMLSup, MathMLIdentifier, MathMLNumber, and MathMLOperator, representing the structure of the equation a² + b² = c²." src="./mathml-a11y-tree.png" 
/>

We could enhance this by adding an <code>aria-label</code> to a wrapper that provides some information about the following formula, especially when it's a well-known one. By using a <code>section</code> with an <code>aria-label</code>, we automatically insert a region into the accessibility tree.

```html
<section aria-label="Pythagorean Theorem">
  <math xmlns="http://www.w3.org/1998/Math/MathML">
    ...
  </math>
</section>
```

> Alternatively, we also have the <a href="https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/math_role"><code>role="math"</code></a> from the ARIA specification. With that, even if you use an image or non-semantic HTML, you can still achieve accessibility. As shown on the <a href="https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/math_role">Mozilla page about <code>role="math"</code></a>, we could have:

Reviewer 2: ```suggestion
Alternatively, we also have the <a href="https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/math_role"><code>"math"</code> role</a> from the ARIA specification. With that, even if you use an image or non-semantic HTML, you can still achieve an accessibile result. As shown on the <a href="https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/math_role">Mozilla page about <code>role="math"</code></a>, we could have:
```

Reviewer: **What exactly does "achieve accessibility"?**

I'm asking because I'm wondering how accessible an image of text is you can't interact with? Also, have you tested the syntax used in the ARIA label with screen readers?

**In which way is this an alternative?** 

“Alternatively” means for me that I can either pick one or the other if I have to. (e.g. `<button>` vs. `<div role="button">`). Visually, for your simple example the ARIA version is an alternative, but for more complex formulas it's not because ARIA doesn't provide any styling. That's worth mentioning.
Also, I'm interested in knowing what screen readers VO/macos, VO/iOS, NVDA/FF, JAWS/Chrome announce for the original and the alternative.

Author: Hi @matuzo ! 
You are right! I reformulated a lot of stuff from the article because I was not also happy about it and I thought I just stayed in the surface, now we have a longer, more detailed article. 

I checked VO/macos, VO/iOS, NVDA/FF and just mentioned what I found about JAWS/Chrome, since I do not have it and could not test with it!

```html
<div role="math" aria-label="a^{2} + b^{2} = c^{2}">
  a<sup>2</sup> + b<sup>2</sup> = c<sup>2</sup>
</div>
```

```html
<img src="pythagorean_theorem.gif" alt="a^{2} + b^{2} = c^{2}" role="math" />
```

## The future of MathML

> Before we look ahead, it's useful to understand where MathML comes from and why there's both a 'Core' and a 'version 4' in development.

Reviewer 2: this might be the curse of knowledge on your part, but i have no idea that there’s a “core” and “version 4”. perhaps this section can be restructured a bit to include some more history and explain this?

Author: but this is exactly what I am doing in the next two paragraphs, right? I just mentioned the existence of the two in this one and explain their history and motivations in the next two ones. Do you need me to further elaborate on them? Or maybe is the way I structured the three paragraphs suggesting they do not have to do with each other and it would be easier if there is just one?

Author: I added a bit more of history and made it everything compact in one more homogeneous paragraph!

As I mentioned at the beginning, the origin of MathML was not the web, it was more of a general-purpose specification for browsers, office suites, computer algebra systems, EPUB readers, and LaTeX-based generators, [as stated in Mozilla](https://developer.mozilla.org/en-US/docs/Web/MathML). MathML Core arose from the need to make it work with web standards, including HTML, CSS, DOM, and JavaScript. Since June 2025, MathML Core has been a [Candidate Recommendation Snapshot](https://www.w3.org/TR/2025/CR-mathml-core-20250624/).

On another note, at the time of this writing, there is [a Working Draft for MathML 4](https://www.w3.org/TR/mathml4/), the next version of MathML. This version aims to be the next "full" spec that extends Core. It keeps the bigger feature set (e.g., Content MathML) and adds, among others, the <code>intent</code> attribute so authors can guide screen-reader speech. With it, we'll be able to do something like this:

```html
<math>
  <mrow intent="equals(power(a,2)+power(b,2),power(c,2))">
    <msup><mi>a</mi><mn>2</mn></msup><mo>+</mo>
    <msup><mi>b</mi><mn>2</mn></msup><mo>=</mo>
    <msup><mi>c</mi><mn>2</mn></msup>
  </mrow>
</math>
```

> <p class="highlight"><strong><abbr title="too long; didn't read">TL;DR:</abbr></strong> MathML Core is what browsers implement today; MathML 4 is the broader language evolving around it..</p>

Reviewer 2: i think this should be at the top of the section. now it’s more of a “too late, did read” 🤓

author_bio: "Accessibility specialist and game developer at [DIE ZEIT](https://www.zeit.de/spiele/index). Currently creating [The Runic Edda](https://www.therunicedda.com), a solo game blending storytelling, Viking history, and fun. I am driven by curiosity, creativity, and the belief that technology should welcome everyone."
date: 2025-12-13
author_links:
  - label: "Web"
    url: "https://www.manuelsanchezdev.com"
    link_label: "manuelsanchezdev.com"
  - label: "Bluesky"
    url: "https://bsky.app/profile/manuelsanchezdev.com"
    link_label: "@manuelsanchezdev.com"
intro: "<p>Math on the web has always been a visual and accessibility challenge. In this article, we learn how to make structures understandable for assistive technologies by using today's MathML Core.</p>"
image: "advent25_13"
---

Maybe it has happened to you that you wanted to write some formulas in HTML to display on a website, and even though there are multiple ways to do it, accessibility is often not considered in the process. How the formula is read by screen readers is crucial to ensure that we don't leave anyone behind. And the main assistive technologies are in different stages, as we will see.

> The web is full of many different and interesting approaches for representing formulas. To name a few, we have TeX/LaTeX source rendered in the browser in different ways, like MathJax or KaTX, we could Unicode math, Canvas/WebGL or even simple PNG/JPG or SVG pictures. However, using native [MathML](https://developer.mozilla.org/en-US/docs/Web/MathML) is usually one of the best options for this task, even if it wasn’t originally designed for the web. Some of its advantages are that it has its own syntax, MathML provides various elements that give the correct semantics to the different parts of a formula, has good screen reader support, requires no JavaScript dependencies and it can be used beyond the browser, as in EPUB or braille/math speech tooling.

Reviewer: ```suggestion
The web is full of many different and interesting approaches for representing formulas. To name a few, we have TeX/LaTeX source rendered in the browser in different ways, like MathJax or KaTeX, we can use Unicode math, Canvas/WebGL, or even simple PNG/JPG or SVG pictures. However, using native [MathML](https://developer.mozilla.org/en-US/docs/Web/MathML) is usually one of the best options for this task, even if it wasn’t initially designed for the web. Some of its advantages are that it has its own syntax, MathML, which provides various elements that give the correct semantics to the different parts of a formula, it has good screen reader support, requires no JavaScript dependencies, and it can be used beyond the browser, as in EPUB or braille/math speech tooling.
```

Author: topi topi

Let's take the famous Pythagorean Theorem as an example.

<section style="margin-bottom: 2rem" aria-labelledby="section-0-heading">
  <h2 id="section-0-heading">Pythagorean Theorem</h2>

>   <p>Next, you have the visual representation of the formula together with the MathML code.</p>

Reviewer: ```suggestion
  <p>The following example is a visual representation of the formula together with the MathML code.</p>
```

Author: top

- <code>math</code> represents the entire mathematical expression.
- <code>msup</code> defines a superscript relationship (a base and an exponent).
- <code>mi</code> is a mathematical identifier, typically a variable such as <strong>a</strong>, <strong>b</strong>, or <strong>c</strong>.
- <code>mn</code> is a mathematical number, like <strong>2</strong>.
- <code>mo</code> is a mathematical operator, such as <strong>+</strong> or <strong>=</strong>.

Other alternatives, like the ones mentioned above, may offer similar capabilities, but they typically rely on an assistive or hidden MathML layer. In practice, MathML remains the only web-standard markup that expresses mathematical roles natively in the DOM.

With this approach, the accessibility tree shows good semantics and VoiceOver knows well what to do.

<img alt="Accessibility tree view of a MathML formula showing nested semantic elements. The tree includes nodes such as MathMLMath, MathMLSup, MathMLIdentifier, MathMLNumber, and MathMLOperator, representing the structure of the equation a² + b² = c²." src="./mathml-a11y-tree.png" 
/>

> However, and this is something we will see through the whole article, screen reader support for the <code>math</code> tag is differently depending on the assistive technology. VoiceOver seems to be doing a pretty good job, [JAWS also makes it easy for both speech and braille](https://www.freedomscientific.com/training/teachers/accessing-math-content-with-jaws-and-fusion/), and [NVDA needs an add-on to make it work called MathCat](https://github.com/nvaccess/nvda/issues/17667) because if not, the <code>math</code> tag will be ignored. A major pull request ([#18323](https://github.com/nvaccess/nvda/pull/18323)) was merged on 17 November 2025 which integrates MathCAT into NVDA core, meaning users won’t have to find/install a separate add-on to handle math.

Reviewer: ```suggestion
However, as we will see throughout the article, screen reader support for the <code>math</code> tag varies across assistive technologies. VoiceOver seems to be doing a pretty good job, [JAWS also makes it easy for both speech and braille](https://www.freedomscientific.com/training/teachers/accessing-math-content-with-jaws-and-fusion/), and [NVDA needs an add-on to make it work called MathCat](https://github.com/nvaccess/nvda/issues/17667) because if not, the <code>math</code> tag will be ignored. A major pull request ([#18323](https://github.com/nvaccess/nvda/pull/18323)) was merged on 17 November 2025 which integrates MathCAT into NVDA core, meaning users won’t have to find/install a separate add-on to handle math.
```

Author: top

<section aria-labelledby="example-formula">
  <h3 id="example-formula">How screen readers interpret the formula</h3>
  <details>
    <summary>NVDA + Firefox (Windows with MathCAT add-on)</summary>

>     region eigh squared plus b squared is equal to c squared space

Reviewer: ```suggestion
    region a squared plus b squared is equal to c squared space
```

Author: This was not a typo, but really what came from NVDA haha. 
But if you want, I can fix this!

Reviewer: Yes, please.

Author: Done!

  </details>
  <details>
    <summary>VoiceOver + Safari (Mac)</summary>
    a squared + b squared = c squared, with 5 items, maths
  </details>
  <details style="margin-bottom: 2rem;">
    <summary>VoiceOver + Safari (iOS)</summary>
    a squared plus b squared equals c squared, Math
  </details>
</section>

> Let's look at a more complicated case. Instead of just the way to display the formula, let's see how to actually prove it and how that will be announced by screen readers.

Reviewer: ```suggestion
Let's look at a more complicated case. Instead of just displaying the formula, let's see how to actually prove it and how screen readers will announce it.
```

Author: like it

- <code>mrow</code>: groups expressions together. For example <code>(a + b)</code> is wrapped in an <code>mrow</code> to indicate that the parentheses and the interior form a single unit before exponentiation.

- <code>mfrac</code>: defines an actual mathematical fraction, not just text with a slash. This allows speech engines to say "one half" instead of "one over two" depending on preferences and locale.

- <code>semantics</code>: this is key. It wraps the expression and lets you attach alternative meanings or encodings. Assistive technologies prefer the first child (your visual MathML), but can fall back to the annotation if needed.

- <code>annotation</code>: stores auxiliary information. In this case, the TeX version of the proof. It does not affect the visual rendering in the browser. Instead, it's metadata for tools that consume MathML, like converters, EPUB readers, or braille translators.

Check out how this is announced by different screen readers!

<section aria-labelledby="example-proof">
  <h3 id="example-proof">How screen readers interpret the proof</h3>
  <details>
  <summary>NVDA + Firefox (Windows with MathCAT add-on)</summary>
  3 lines

>   line 1 open paren eigh plus b close paren squared is equal to c squared plus 4 times 1 half eigh b

Reviewer: Please check each output to see if it makes sense and fix bugs, e.g., _eigh_ instead of _a_.

Author: Same as the [comment here](https://github.com/matuzo/HTMHell/pull/204#discussion_r2571551189), I can fix it, but that is the exact output given by NVDA. I can specify `Speech Viewer` in all the cases or fix it, you tell me :)

Reviewer: Please fix it, because it doesn't make sense and confuses readers

Author: Sure!

## Some A11y Enhancements

We could enhance this by adding an <code>aria-label</code> to a wrapper that provides some information about the following formula, especially when it's a well-known one. By using a <code>section</code> with an <code>aria-label</code> or <code>aria-labelledby</code> together with another element giving the accessible name, we automatically insert a region into the accessibility tree.

```html
<section aria-labelledby="section-1-heading">
  <h2 id="section-1-heading">Pythagorean Theorem</h2>
  <math xmlns="http://www.w3.org/1998/Math/MathML"> ... </math>
</section>
```

Also, for users who zoom the browser up to 400%, we might want to add a `max-width: 100%` and `overflow-x: auto`, so that the formula remains readable, does not break the page and we allow horizontal scrolling only inside the math block, and not at the entire page level.

## Conveying mathematical meaning with ARIA

> As an alternative to using MathML to convey mathematical meaning in simple examples, we also have the <a href="https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/math_role"><code>math</code> role</a> from the ARIA specification. With that, we can communicate the mathematical semantics even when we rely on images or non-semantic HTML. However, it does not tend to give good results with VoiceOver on macOS, for example.

Reviewer: Is something that doesn't work in some screen readers really an alternative?

Author: Valid point, I will remove the part of the alternative!

> As an alternative to using MathML to convey mathematical meaning in simple examples, we also have the <a href="https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/math_role"><code>math</code> role</a> from the ARIA specification. With that, we can communicate the mathematical semantics even when we rely on images or non-semantic HTML. However, it does not tend to give good results with VoiceOver on macOS, for example.

Reviewer: > it does not tend to give good results with VoiceOver on macOS

tend is a bit too soft. It doesn't just tend to not give good results, it doesn't give good results.

Author: yes!

As shown on the <a href="https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/math_role">MDN page for the <code>math</code> role</a>, we could have:

```html
<div role="math" aria-label="a^{2} + b^{2} = c^{2}">
  a<sup>2</sup> + b<sup>2</sup> = c<sup>2</sup>
</div>
```

<section aria-labelledby="example-div">
  <h3 id="example-div">Markup with a div with the math role and how screen readers interpret it</h3>
  <details>
    <summary>Markup</summary>

>     <div style="margin-bottom: 2rem;" role="math" aria-label="a^{2} + b^{2} = c^{2}">

Reviewer: > a^{2} + b^{2} = c^{2}

Where does this syntax come from? Is this like an official notation? (sorry I'm bad at Maths)

Author: I am probable worse than you haha. 
Just took it from here: 
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/math_role

Reviewer: Okay, I understand.

      a<sup>2</sup> + b<sup>2</sup> = c<sup>2</sup>
    </div>
  </details>
  <details>
    <summary>NVDA + Firefox (Windows with MathCAT add-on)</summary>

>   just announce the markup, line by line

Reviewer: line by line? There is only one line!?

Author: Noo, in this case it really reads the markup of the three lines (one by one):

```
<div style="margin-bottom: 2rem;" role="math" aria-label="a^{2} + b^{2} = c^{2}">
      a<sup>2</sup> + b<sup>2</sup> = c<sup>2</sup>
</div>
```

Reviewer: What do you mean by line? Do you mean character by character?

Author: I mean that it reads it like this
<div style="margin-bottom: 2rem;" role="math" aria-label="a^{2} + b^{2} = c^{2}">
then
 a<sup>2</sup> + b<sup>2</sup> = c<sup>2</sup>
 finally
 </div>
 
 But I can just remove the line by line from the text and that's it!

Reviewer: A okay. Then maybe change it to "Just announces the text"!?

Author: Deal!

    not read, just announces "with 6 items, maths"
  </details>
  <details style="margin-bottom: 2rem;" >
    <summary>VoiceOver + Safari (iOS)</summary>
    a caret left curly bracket 2 right curly bracket plus b caret left curly bracket 2 right curly bracket equals c caret left curly bracket, Math
  </details>
</section>

```html
<img src="pythagorean_theorem.png" alt="a^{2} + b^{2} = c^{2}" role="math" />
```

<section aria-labelledby="example-img">
  <h3 id="example-img">Markup with a img with the math role and how screen readers interpret it</h3>
  <details>
    <summary>Markup</summary>
    <img width="150" alt="a^{2} + b^{2} = c^{2}" src="./pythagorean-theorem.png" role="math" />
  </details>
  <details>
    <summary>NVDA + Firefox (Windows with MathCAT add-on)</summary>
  just announce the markup
  </details>
  <details>
    <summary>VoiceOver + Safari (Mac)</summary>
    not read, just announces "maths"
  </details>
  <details style="margin-bottom: 2rem;">
    <summary>VoiceOver + Safari (iOS)</summary>
    a caret left curly bracket 2 right curly bracket plus b caret left curly bracket 2 right curly bracket equals c caret left curly bracket, Math
  </details>
</section>

> In practice, using the math role helps assistive technologies understand that the content is mathematical, but it still doesn’t provide enough semantic detail for them to announce the expression as accurately as MathML does.

Reviewer: Callback to a previous comment to emphasise it: Here you say that it's not practical, which means it's not an alternative.

Author: True, it is not an alternative (yet)!

Author: I removed the part with the alternative!

## The future of MathML

<p class="highlight"><strong><abbr title="too long; didn't read">TL;DR:</abbr></strong> MathML Core is what browsers implement today; MathML 4 is the broader language evolving around it.</p>

> First, we are going to focus on where MathML comes from and why there's both a 'Core' and a 'version 4' in development. As I mentioned at the beginning, the origin of MathML was not the web, it was more of a general-purpose specification for browsers, office suites, computer algebra systems, EPUB readers, and LaTeX-based generators, [as stated in Mozilla](https://developer.mozilla.org/en-US/docs/Web/MathML). MathML Core arose from the need to make it work with web standards, including HTML, CSS, DOM, and JavaScript. Historically, the full MathML spec was broad and partly underspecified for browsers, which led to uneven or incomplete implementations across engines. MathML Core therefore narrows the language to the subset that can be precisely defined on top of the Web Platform, improving testability and cross-browser interoperability. Since June 2025, MathML Core has been a [Candidate Recommendation Snapshot](https://www.w3.org/TR/2025/CR-mathml-core-20250624/). On another note, at the time of this writing, there is [a Working Draft for MathML 4](https://www.w3.org/TR/mathml4/), the next version of MathML. This version aims to be the next "full" spec that extends Core. It keeps the larger feature set (e.g., Content MathML) and adds, among others, the <code>intent</code> attribute so authors can guide screen-reader speech. With it, we'll be able to do something like this:

Reviewer: ```suggestion
As I mentioned at the beginning, the origin of MathML was not the web, it was more of a general-purpose specification for browsers, office suites, computer algebra systems, EPUB readers, and LaTeX-based generators, [as stated in Mozilla](https://developer.mozilla.org/en-US/docs/Web/MathML). MathML Core arose from the need to make it work with web standards, including HTML, CSS, DOM, and JavaScript. Historically, the full MathML spec was broad and partly underspecified for browsers, which led to uneven or incomplete implementations across engines. MathML Core therefore narrows the language to the subset that can be precisely defined on top of the Web Platform, improving testability and cross-browser interoperability. Since June 2025, MathML Core has been a [Candidate Recommendation Snapshot](https://www.w3.org/TR/2025/CR-mathml-core-20250624/). On another note, at the time of this writing, there is [a Working Draft for MathML 4](https://www.w3.org/TR/mathml4/), the next version of MathML. This version aims to be the next "full" spec that extends Core. It keeps the larger feature set (e.g., Content MathML) and adds, among others, the <code>intent</code> attribute so authors can guide screen-reader speech. With it, we'll be able to do something like this:
```

Author: done!
